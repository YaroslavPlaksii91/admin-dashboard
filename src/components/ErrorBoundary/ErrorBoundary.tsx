import { Component, ErrorInfo, ReactNode } from 'react';

import { ErrorBoundaryProps, ErrorBoundaryState } from './types';

export class ErrorBoundary extends Component<
  ErrorBoundaryProps,
  ErrorBoundaryState
> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false, error: null };
  }

  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
    return { hasError: true, error };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    console.error(error, errorInfo);
  }

  render(): ReactNode {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Something went wrong!</h1>
          <p>{this.state.error?.toString()}</p>
        </div>
      );
    }

    return this.props.children;
  }
}
