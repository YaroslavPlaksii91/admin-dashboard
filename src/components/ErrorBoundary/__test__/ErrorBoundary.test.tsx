import { render, screen, cleanup } from '@testing-library/react';

import { ErrorBoundary } from '../ErrorBoundary';

describe('ErrorBoundary component', () => {
  afterEach(cleanup);

  it('makes snapshot', () => {
    const { asFragment } = render(
      <ErrorBoundary>
        <div>Child Component</div>
      </ErrorBoundary>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders children when there is no error', () => {
    render(
      <ErrorBoundary>
        <div>Child Component</div>
      </ErrorBoundary>,
    );

    const childElement = screen.getByText('Child Component');

    expect(childElement).toBeInTheDocument();
  });

  it('renders error message when there is an error', () => {
    const ErrorThrower = () => {
      throw new Error('Test Error');
    };

    render(
      <ErrorBoundary>
        <ErrorThrower />
      </ErrorBoundary>,
    );

    const errorMessage = screen.getByText('Something went wrong!');

    expect(errorMessage).toBeInTheDocument();
  });
});
