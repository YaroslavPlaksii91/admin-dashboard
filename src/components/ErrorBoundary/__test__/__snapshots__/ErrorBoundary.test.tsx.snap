// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component makes snapshot 1`] = `
<DocumentFragment>
  <div>
    Child Component
  </div>
</DocumentFragment>
`;
